packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -6636,15 +6632,31 @@ class RepaintBoundary extends SingleChildRenderObjectWidget {
 /// ** See code in examples/api/lib/widgets/basic/ignore_pointer.0.dart **
 /// {@end-tool}
 ///
+/// ## Semantics
+///
+/// Using this widget may also affect how semantics subtree underneath this
+/// widget is collected.
+///
+/// {@template flutter.widgets.IgnorePointer.Semantics}
+/// If [ignoringSemantics] is true, the semantics subtree is dropped.
+///
+/// If [ignoringSemantics] is false, the semantics subtree is collected as
```

Comment:
Yes, the SemanticsDebugger want to covert all pointer action into semantics action and dispatch it to the subtree. It does this by adding its own gesture recognizer and add the ignore pointer to the subtree. Since it needs to send the semantics action to the subtree, so it don't want those semantics action to be blocked.

Thus it needs
IgnorePointer(
ignoringSemantics: false
)

I think WidgetInspector has the same use case, it want to block pointer event to the subtree but still display the full semantics tree when user inspecting the widget.